app/Console/Commands/CreateMasterVoucherLogReport.php

```diff
@@ -0,0 +1,308 @@
+        'Payment Request Date',
+        'Trader Name',
+        'Retail Outlet',
+        'Reimbursed Date'
+    ];
+
+    /**
+     * The report's query template
+     *
+     * TODO: refactor this as eloquent lookups when it's finalised.
+     *      This will eventually help with the fact we'll need to chunk data too.
+     * @var string $report
+     */
+    private $report = <<<EOD
+SELECT
+  sponsors.name AS 'Area',
+  vouchers.code AS 'Voucher Code',
+  dispatch_date AS 'Dispatch Date',
+  disbursed_at AS 'Disbursed Date',
+  rvid AS 'RVID',
+  pri_carer_name AS 'Participant',
+  payment_request_date AS 'Payment Request Date',
+  trader_name AS 'Trader Name',
+  market_name AS 'Retail Outlet',
+  reimbursed_date AS 'Reimbursed Date'
+
+FROM vouchers
+
+  # Join for each voucher\'s sponsor name
+  LEFT JOIN sponsors ON vouchers.sponsor_id = sponsors.id
+
+  # Get our trader and market names.
+  LEFT JOIN (
+    SELECT traders.id,
+           traders.name AS trader_name,
+           markets.name AS market_name
+    FROM traders
+    LEFT JOIN markets ON traders.market_id = markets.id
+  ) AS markets_query
+    ON markets_query.id = vouchers.trader_id
+
+  # Pivot dispatched date
+  LEFT JOIN (
+    SELECT voucher_states.voucher_id,
+           voucher_states.created_at AS dispatch_date
+    FROM voucher_states
+    WHERE voucher_states.`to` = 'dispatched'
+  ) AS dispatch_query
+    ON dispatch_query.voucher_id = vouchers.id
+
+  # Pivot payment_request date
+  LEFT JOIN (
+    SELECT voucher_states.voucher_id,
+           voucher_states.created_at AS payment_request_date
+    FROM voucher_states
+    WHERE voucher_states.`to` = 'payment_pending'
+  ) AS payment_request_query
+    ON payment_request_query.voucher_id = vouchers.id
+
+  # Pivot reimbursed date
+  LEFT JOIN (
+    SELECT voucher_states.voucher_id,
+           voucher_states.created_at AS reimbursed_date
+    FROM voucher_states
+    WHERE voucher_states.`to` = 'reimbursed'
+  ) AS reimburse_query
+    ON reimburse_query.voucher_id = vouchers.id
+
+  # Get fields relevant to bundles (pri_carer/RVID/disbursed_at)
+  LEFT JOIN (
+    SELECT bundles.id,
+           bundles.disbursed_at AS disbursed_at,
+           # LPAD will truncate values over 4 characters (or 9999 rvids).
+           CONCAT(centres.prefix, LPAD(families.centre_sequence, 4, 0)) AS rvid,
+           pri_carer_query.name as pri_carer_name
+    FROM bundles
+      LEFT JOIN registrations on bundles.registration_id = registrations.id
+      LEFT JOIN families ON registrations.family_id = families.id
+      LEFT JOIN centres ON families.initial_centre_id = centres.id
+        
+      # Need to join the Primary Carer here; Primary Carers are only relevant via bundles.
+      LEFT JOIN (
+        
+        # We need the *first*, by self join grouping (classic technique, so SQLite can cope)
+        SELECT t1.name, t1.family_id
+        FROM carers t1
+        INNER JOIN (
+          SELECT MIN(id) as id
```

Comment:
Will soft deletes or reinstated main carers be an issue here? I can't remember how you settled on designating Main Carer.

Reply:
Sounds good to me.